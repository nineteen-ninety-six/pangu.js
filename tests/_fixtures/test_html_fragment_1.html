<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <title>Test HTML Fragment 1 - Text Node Splitting</title>
</head>
<body>
    <div id="test-container">
        <p>这是一个测试段落，包含了"引号内容"和其他文字。</p>
        
        <!-- Reproducing the exact structure from the browser console -->
        <div id="split-text-nodes">
            <!-- Text nodes split around quotes as shown in console -->
            <span>社</span>"<span>DF</span>
        </div>
        
        <!-- Another example with multiple splits -->
        <div id="complex-split">
            前面的文字"<span>中间的内容</span>"后面的文字
        </div>
        
        <!-- Simulating AJAX/dynamic content that might cause this -->
        <div id="dynamic-content"></div>
    </div>

    <script>
        // Simulate dynamic content insertion that might split text nodes
        document.addEventListener('DOMContentLoaded', function() {
            const dynamicDiv = document.getElementById('dynamic-content');
            
            // Method 1: Direct innerHTML that might get parsed differently
            dynamicDiv.innerHTML = '动态内容"<span>测试</span>"结束';
            
            // Method 2: Creating text nodes programmatically
            const container2 = document.createElement('div');
            container2.appendChild(document.createTextNode('程序生成的'));
            container2.appendChild(document.createTextNode('"'));
            const span = document.createElement('span');
            span.textContent = '内容';
            container2.appendChild(span);
            container2.appendChild(document.createTextNode('"'));
            container2.appendChild(document.createTextNode('文字'));
            document.body.appendChild(container2);
        });
    </script>
</body>
</html>