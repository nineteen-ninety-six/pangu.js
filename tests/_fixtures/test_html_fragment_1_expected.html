<!DOCTYPE html>
<html lang="zh-CN">
<head>
    <meta charset="UTF-8">
    <title>Expected Result - After Pangu.js Processing</title>
    <style>
        .highlight { background-color: yellow; }
    </style>
</head>
<body>
    <h2>Expected Results After Pangu.js Processing</h2>
    
    <!-- Expected: 社 "DF -->
    <div class="test-case">
        <h3>Test 1: 社"DF → 社 "DF</h3>
        <div>
            <span>社</span><span class="highlight"> "</span><span>DF</span>
        </div>
    </div>

    <!-- Expected: 前面的文字 "中间的内容" 后面的文字 -->
    <div class="test-case">
        <h3>Test 2: Complex Quote Structure</h3>
        <div>
            前面的文字<span class="highlight"> "</span><span>中间的内容</span><span class="highlight">" </span>后面的文字
        </div>
    </div>

    <!-- Expected: 【UCG 中字】"數毛社" DF 的《戰神 4》全新演示解析 -->
    <div class="test-case">
        <h3>Test 3: Full Example from Issue</h3>
        <div>
            【UCG<span class="highlight"> </span>中字】<span class="highlight">"</span>數毛社<span class="highlight">" </span>DF<span class="highlight"> </span>的《戰神<span class="highlight"> </span>4》全新演示解析
        </div>
    </div>

    <script>
        // Visual demonstration of what should happen
        document.querySelectorAll('.highlight').forEach(el => {
            el.title = 'Space added by pangu.js';
        });
    </script>
</body>
</html>